WooCommerce/Classes/ViewRelated/Blaze/BlazeCampaignList/BlazeCampaignListView.swift

```diff
@@ -0,0 +1,140 @@
+import SwiftUI
+import struct Yosemite.Site
+
+/// Blaze campaign entry points.
+enum BlazeCampaignSource: String {
+    /// From the Blaze section on My Store tab.
+    case myStoreSection = "my_store_section"
+    /// From the Blaze campaign list
+    case campaignList = "campaign_list"
+}
+
+/// Hosting controller for `BlazeCampaignListView`
+///
+final class BlazeCampaignListHostingController: UIHostingController<BlazeCampaignListView> {
+
+    init(site: Site, viewModel: BlazeCampaignListViewModel) {
+        super.init(rootView: BlazeCampaignListView(viewModel: viewModel, siteURL: site.url.trimHTTPScheme()))
+
+        rootView.onCreateCampaign = { [weak self] in
+            let webViewModel = BlazeWebViewModel(source: .campaignList, site: site, productID: nil) {
+                viewModel.loadCampaigns()
+            }
+            let webViewController = AuthenticatedWebViewController(viewModel: webViewModel)
+            self?.navigationController?.show(webViewController, sender: self)
+        }
+    }
+
+    @available(*, unavailable)
+    required dynamic init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+/// View for showing a list of campaigns.
+///
+struct BlazeCampaignListView: View {
+    @ObservedObject private var viewModel: BlazeCampaignListViewModel
+    @State private var selectedCampaignURL: URL?
+
+    var onCreateCampaign: () -> Void = {}
+
+    private let siteURL: String
+
+    init(viewModel: BlazeCampaignListViewModel, siteURL: String) {
+        self.viewModel = viewModel
+        self.siteURL = siteURL
+    }
+
+    var body: some View {
+        Group {
+            switch viewModel.syncState {
+            case .results:
+                RefreshableInfiniteScrollList(spacing: Layout.contentSpacing,
```

Comment:
Nit: What do you think about separating the empty view, loading view and the list content into private variables for readability?

Reply:
Since the view is not too complicated, I prefer keeping everything in one place to avoid having to scroll up and down to view the content. I think we can have different preferred styles, and one can pick whatever they are more comfortable with.